lib/core.ps1

```diff
@@ -589,9 +589,18 @@ function Invoke-ExternalCommand {
 }
 
 function env($name,$global,$val='__get') {
-    $target = 'User'; if($global) {$target = 'Machine'}
-    if($val -eq '__get') { [environment]::getEnvironmentVariable($name,$target) }
-    else { [environment]::setEnvironmentVariable($name,$val,$target) }
+    $target = 'User'
+    $EnvironmentRegisterKey = 'HKCU:\Environment'
+    if ($global) {
+        $target = 'Machine'
+        $EnvironmentRegisterKey = 'HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment'
+    }
+    if ($val -eq '__get') {
+        (Get-Item -Path $EnvironmentRegisterKey).
+            GetValue($name, '', [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames)
+    } else {
+        [environment]::setEnvironmentVariable($name,$val,$target)
```

Comment:
We can use `Set-Item -Path $EnvironmentRegisterKey` here as well for uniformity.

Reply:
At first I thought there was no difference in setting the environment variable with the previous method or with `Set-Item` etc, so it might be better not to change the code.
However when I modified the code, I found that if I used `[environment]::setEnvironmentVariable($name, $val, $target)`, the corresponding environment variable type would become `REG_SZ`. This can be seen in detail in `regedit`.

This not only defeats my original intention(avoid `%` substitutions in env), but also leads to some unusable situations.
Because the `%%` variable in the `REG_SZ` type environment variable will not be replaced with the corresponding content when using. And it needs to use the `REG_EXPAND_SZ` type to make it work.
So I used `-Type 'ExpandString'` to make it correct, refer to [Microsoft's documentation](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/set-itemproperty?view=powershell-7.2#-type).

I installed Python on another machine with my modified code and it worked fine.

Digression: When I want to use `Get-ItemProperty` to keep correspondence with `Set-ItemProperty`, I find that I can't get unexpanded environment variables through it.([an answer from stackoverflow](https://stackoverflow.com/questions/51776242/how-to-retrieve-original-unresolved-unexpanded-registry-value)). There may be other better ways, but this will work and be consistent to a certain extent.

Reply:
> Digression: When I want to use `Get-ItemProperty` to keep correspondence with `Set-ItemProperty`, I find that I can't get unexpanded environment variables through it.([an answer from stackoverflow](https://stackoverflow.com/questions/51776242/how-to-retrieve-original-unresolved-unexpanded-registry-value)). There may be other better ways, but this will work and be consistent to a certain extent.

Hmm, what about using [`[Microsoft.Win32.RegistryKey].GetValue()`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.win32.registrykey.getvalue?view=netframework-4.7.2) and [`[Microsoft.Win32.RegistryKey].SetValue()`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.win32.registrykey.setvalue?view=netframework-4.7.2), as the answer mentioned?

Reply:
Haha I accidentally missed the link in the answer.

Calling C# code in Powershell is too difficult for me, but I still have some discoveries in my study.

`[Microsoft.Win32.Registry]::CurrentUser.OpenSubKey('Environment', $true).SetValue('Foo','Bar')` is used to set user environment variables. The system's environment variable setting method will be longer.

When using `(Get-Item -Path $EnvironmentRegisterKey).SetValue('Foo','Bar')`, it will prompt "Cannot write to the registry key.". So the object obtained in this way is read-only, and there is no way to use the `SetValue()` method.

So I combined both ways. Sample code: `(Get-Item -Path 'HKCU:').OpenSubKey('Environment', $true).SetValue('Foo', 'Bar')`

So I modified my code and it works fine now!

Reply:
Great! I've pushed a commit to simplify variables a bit, let me know if it's okay.

Reply:
Of course！